contracts/metatxToken/MetaTxToken.sol

```diff
@@ -0,0 +1,269 @@
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.7.3;
+
+contract MetaTxToken is DSTokenBaseMeta(0), DSAuthMeta {
+  uint8 public decimals;
+  string public symbol;
+  string public name;
+
+  bool public locked;
+  bytes32 public DOMAIN_SEPARATOR;
+
+  mapping(address => uint256) metatransactionNonces;
+
+  function getMetatransactionNonce(address _user) override public view returns (uint256 nonce){
+    return metatransactionNonces[_user];
+  }
+
+  function incrementMetatransactionNonce(address _user) override internal {
+    metatransactionNonces[_user] = add(metatransactionNonces[_user], 1);
+  }
+
+  event Mint(address indexed guy, uint wad);
+  event Burn(address indexed guy, uint wad);
+
+  modifier unlocked {
+    if (locked) {
+      require(isAuthorized(msgSender(), msg.sig), "colony-token-unauthorised");
+    }
+    _;
+  }
+
+  constructor(string memory _name, string memory _symbol, uint8 _decimals) {
+    name = _name;
+    symbol = _symbol;
+    decimals = _decimals;
+    locked = true;
+
+    uint chainId;
+    assembly {
+        chainId := chainid()
+    }
+
+    DOMAIN_SEPARATOR = keccak256(
+        abi.encode(
+            keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)"),
+            keccak256(bytes(name)),
+            keccak256(bytes("1")),
+            chainId,
+            address(this)
+        )
+    );
+  }
+
+  function transferFrom(address src, address dst, uint wad) public
+  unlocked override
+  returns (bool)
+  {
+    return super.transferFrom(src, dst, wad);
+  }
+
+  function mint(uint wad) public auth {
+    mint(msgSender(), wad);
+  }
+
+  function burn(uint wad) public {
+    burn(msgSender(), wad);
+  }
+
+  function mint(address guy, uint wad) public auth {
+    _balances[guy] = add(_balances[guy], wad);
+    _supply = add(_supply, wad);
+    emit Mint(guy, wad);
+    emit Transfer(address(0x0), guy, wad);
+  }
+
+  function burn(address guy, uint wad) public {
+    if (guy != msgSender()) {
+      require(_approvals[guy][msgSender()] >= wad, "ds-token-insufficient-approval");
+      _approvals[guy][msgSender()] = sub(_approvals[guy][msgSender()], wad);
+    }
+
+    require(_balances[guy] >= wad, "ds-token-insufficient-balance");
+    _balances[guy] = sub(_balances[guy], wad);
+    _supply = sub(_supply, wad);
+    emit Burn(guy, wad);
+  }
+
+  function unlock() public
+  auth
+  {
+    locked = false;
+  }
+
+  // Pinched from https://github.com/Uniswap/uniswap-v2-core/blob/master/contracts/UniswapV2ERC20.sol
+  // Which is also licenced under GPL V3
+
+  // keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)");
+  bytes32 public constant PERMIT_TYPEHASH = 0x6e71edae12b1b97f4d1f60370fef10105fa2faae0126114a169c64845d6126c9;
```

Comment:
Why not just set the constant to the hash result directly?

Reply:
This came from the existing implementation, so I have no skin in this particular game, but I believe this is cheaper gas-wise.